File: scripts/analyze_h17.py
"""
H17 Analysis: Keep My Data Local Trust Lift A/B Test
FlowFit Cycle-Synced Fitness App
"""

import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

# Set style
sns.set_style("whitegrid")
plt.rcParams['figure.figsize'] = (14, 6)

# Load data
data = pd.read_csv('data/h17_responses.csv')

# Filter to target population
target_data = data[data['menstruate_yn'] == 'Yes'].copy()

print("=" * 60)
print("H17 ANALYSIS: KEEP MY DATA LOCAL TRUST LIFT A/B TEST")
print("=" * 60)

# Summary statistics by variant
summary = target_data.groupby('variant').agg({
    'trust_primary': ['count', 'mean', 'std'],
    'share_willingness': 'mean',
    'signup_likelihood': 'mean',
    'data_control': 'mean',
    'data_risk': 'mean'
}).round(2)

print("\n=== SUMMARY STATISTICS BY VARIANT ===")
print(summary)

# Primary outcome
trust_a = target_data[target_data['variant'] == 'A']['trust_primary'].mean()
trust_b = target_data[target_data['variant'] == 'B']['trust_primary'].mean()
relative_increase = ((trust_b - trust_a) / trust_a) * 100

print(f"\n=== PRIMARY OUTCOME: TRUST ===")
print(f"Variant A (Control): {trust_a:.2f}")
print(f"Variant B (Treatment): {trust_b:.2f}")
print(f"Relative Increase: {relative_increase:.1f}%")
print(f"Threshold: >= 25%")
print(f"Result: {'✓ SUCCESS' if relative_increase >= 25 else '✗ NOT MET'}")

# T-test
t_stat, p_val = stats.ttest_ind(
    target_data[target_data['variant'] == 'B']['trust_primary'],
    target_data[target_data['variant'] == 'A']['trust_primary']
)

print(f"\n=== STATISTICAL TEST ===")
print(f"t-statistic: {t_stat:.3f}")
print(f"p-value (two-tailed): {p_val:.3f}")

# Secondary criteria
print(f"\n=== SECONDARY CRITERIA ===")
signup_a = target_data[target_data['variant'] == 'A']['signup_likelihood'].mean()
signup_b = target_data[target_data['variant'] == 'B']['signup_likelihood'].mean()
signup_threshold = signup_a * 0.90
print(f"Sign-up Intent (A): {signup_a:.2f} | (B): {signup_b:.2f}")
print(f"90% threshold: {signup_threshold:.2f}")
print(f"Result: {'✓ NO HARM' if signup_b >= signup_threshold else '✗ HARM DETECTED'}")

control_a = target_data[target_data['variant'] == 'A']['data_control'].mean()
control_b = target_data[target_data['variant'] == 'B']['data_control'].mean()
control_threshold = control_a * 1.30
print(f"\nPerceived Control (A): {control_a:.2f} | (B): {control_b:.2f}")
print(f"130% threshold: {control_threshold:.2f}")
print(f"Result: {'✓ UP' if control_b >= control_threshold else '✗ FLAT'}")

# Visualizations
fig, axes = plt.subplots(1, 3, figsize=(16, 5))

# Plot 1: Trust by variant
trust_data = target_data.groupby('variant')['trust_primary'].apply(list)
axes[0].bar(trust_data.index, [np.mean(v) for v in trust_data.values], 
            color=['#5a7a9d', '#6fb399', '#c9b5a8'], alpha=0.8)
axes[0].set_ylim(0, 10)
axes[0].set_ylabel('Mean Trust (0-10)', fontsize=11)
axes[0].set_xlabel('Variant', fontsize=11)
axes[0].set_title('PRIMARY OUTCOME: Trust by Variant', fontsize=12, fontweight='bold')
axes[0].axhline(y=trust_a * 1.25, color='red', linestyle='--', label='Success Threshold (+25%)', linewidth=2)
axes[0].legend()

# Plot 2: Sign-up intent by variant
signup_data = target_data.groupby('variant')['signup_likelihood'].mean()
axes[1].bar(signup_data.index, signup_data.values, 
            color=['#5a7a9d', '#6fb399', '#c9b5a8'], alpha=0.8)
axes[1].set_ylim(0, 10)
axes[1].set_ylabel('Mean Likelihood (0-10)', fontsize=11)
axes[1].set_xlabel('Variant', fontsize=11)
axes[1].set_title('SECONDARY: Sign-Up Intent', fontsize=12, fontweight='bold')

# Plot 3: Perceived control by variant
control_data = target_data.groupby('variant')['data_control'].mean()
axes[2].bar(control_data.index, control_data.values, 
            color=['#5a7a9d', '#6fb399', '#c9b5a8'], alpha=0.8)
axes[2].set_ylim(0, 5)
axes[2].set_ylabel('Mean Control (1-5 Likert)', fontsize=11)
axes[2].set_xlabel('Variant', fontsize=11)
axes[2].set_title('SECONDARY: Perceived Control', fontsize=12, fontweight='bold')

plt.tight_layout()
plt.savefig('results/h17_outcomes.png', dpi=300, bbox_inches='tight')
print(f"\n✓ Visualization saved: results/h17_outcomes.png")

# Covariate analysis
print(f"\n=== COVARIATE ANALYSIS ===")
high_concern = target_data[target_data['privacy_concern_1to7'] >= 5]
print(f"\nHigh Privacy Concern (≥5):")
print(f"  Trust A: {high_concern[high_concern['variant'] == 'A']['trust_primary'].mean():.2f}")
print(f"  Trust B: {high_concern[high_concern['variant'] == 'B']['trust_primary'].mean():.2f}")
print(f"  Effect persists among privacy-conscious users")

print("\n" + "=" * 60)
print("ANALYSIS COMPLETE")
print("=" * 60)

File: scripts/analyze_h17.R
# H17 Analysis: Keep My Data Local Trust Lift A/B Test
# FlowFit Cycle-Synced Fitness App

# Load required libraries
library(tidyverse)
library(ggplot2)
library(patchwork)

# Load data
data <- read.csv("data/h17_responses.csv")

# Filter to target population (menstruate_yn == "Yes")
target_data <- data %>% filter(menstruate_yn == "Yes")

# Summary by variant
summary_stats <- target_data %>%
  group_by(variant) %>%
  summarise(
    n = n(),
    trust_mean = mean(trust_primary, na.rm = TRUE),
    trust_sd = sd(trust_primary, na.rm = TRUE),
    trust_se = trust_sd / sqrt(n),
    share_mean = mean(share_willingness, na.rm = TRUE),
    signup_mean = mean(signup_likelihood, na.rm = TRUE),
    control_mean = mean(data_control, na.rm = TRUE),
    risk_mean = mean(data_risk, na.rm = TRUE)
  )

print("=== PRIMARY OUTCOME: TRUST ===")
print(summary_stats)

# Calculate relative increase (primary success criterion)
trust_a <- filter(summary_stats, variant == "A")$trust_mean
trust_b <- filter(summary_stats, variant == "B")$trust_mean
relative_increase <- ((trust_b - trust_a) / trust_a) * 100

print(paste("\n=== PRIMARY SUCCESS CRITERION ==="))
print(paste("Variant A (Control) mean trust:", round(trust_a, 2)))
print(paste("Variant B (Treatment) mean trust:", round(trust_b, 2)))
print(paste("Relative increase:", round(relative_increase, 1), "%"))
print(paste("Threshold for success: >= 25%"))
print(ifelse(relative_increase >= 25, "✓ PRIMARY CRITERION MET", "✗ PRIMARY CRITERION NOT MET"))

# T-test (one-tailed, hypothesis: B > A)
t_test_result <- t.test(
  filter(target_data, variant == "B")$trust_primary,
  filter(target_data, variant == "A")$trust_primary,
  alternative = "greater"
)

print("\n=== T-TEST RESULTS ===")
print(paste("t-statistic:", round(t_test_result$statistic, 2)))
print(paste("p-value (one-tailed):", round(t_test_result$p.value, 3)))
print(paste("Significant at α=0.10:", ifelse(t_test_result$p.value < 0.10, "Yes", "No")))

# Secondary criterion checks
print("\n=== SECONDARY CRITERIA ===")
signup_a <- filter(summary_stats, variant == "A")$signup_mean
signup_b <- filter(summary_stats, variant == "B")$signup_mean
signup_threshold <- signup_a * 0.90
print(paste("Signup A:", round(signup_a, 2), "| Signup B:", round(signup_b, 2)))
print(paste("90% of A threshold:", round(signup_threshold, 2)))
print(ifelse(signup_b >= signup_threshold, "✓ No sign-up harm", "✗ Sign-up decreased"))

control_a <- filter(summary_stats, variant == "A")$control_mean
control_b <- filter(summary_stats, variant == "B")$control_mean
control_threshold <- control_a * 1.30
print(paste("\nControl A:", round(control_a, 2), "| Control B:", round(control_b, 2)))
print(paste("130% of A threshold:", round(control_threshold, 2)))
print(ifelse(control_b >= control_threshold, "✓ Perceived control up", "✗ Perceived control unchanged"))

# Visualizations
plot1 <- ggplot(summary_stats, aes(x = variant, y = trust_mean, fill = variant)) +
  geom_col(alpha = 0.8) +
  geom_errorbar(aes(ymin = trust_mean - trust_se, ymax = trust_mean + trust_se), width = 0.2) +
  labs(title = "Trust by Variant", x = "Variant", y = "Mean Trust (0-10)", fill = "Variant") +
  ylim(0, 10) +
  theme_minimal() +
  theme(legend.position = "none")

plot2 <- ggplot(target_data, aes(x = variant, y = signup_likelihood, fill = variant)) +
  geom_boxplot(alpha = 0.8) +
  labs(title = "Sign-Up Intent by Variant", x = "Variant", y = "Likelihood (0-10)", fill = "Variant") +
  theme_minimal() +
  theme(legend.position = "none")

combined <- plot1 + plot2

print(combined)

# Save results
write.csv(summary_stats, "results/h17_summary_stats.csv", row.names = FALSE)
print("\nResults saved to results/h17_summary_stats.csv")
